orttraining/orttraining/training_api/checkpoint.cc

```diff
@@ -1,581 +1,720 @@
+                    ONNX_NAMESPACE::ModelProto& model_proto) {
+  std::vector<uint8_t> checkpoint_bytes;
+  ORT_RETURN_IF_ERROR(Load::FromFile(checkpoint_path, checkpoint_bytes));
+
+  const auto* fbs_checkpoint = fbs::GetCheckpoint(checkpoint_bytes.data());
+  ORT_RETURN_IF_NOT(fbs_checkpoint, "Checkpoint is invalid. Expected: Valid checkpoint flatbuffer. Actual: nullptr.");
+
+  const auto checkpoint_version = fbs_checkpoint->version();
+  ORT_RETURN_IF_NOT(IsCheckpointVersionSupported(checkpoint_version),
+                    "Checkpoint is invalid. Expected: Checkpoint version ", checkpoint_version, " is not supported.");
+
+  auto* module_state = fbs_checkpoint->module_state();
+  if (nullptr == module_state) {
     return Status::OK();
   }
 
-  for (auto& property_file_path : custom_property_filenames) {
-    std::vector<ONNX_NAMESPACE::TensorProto> property_protos{};
-    auto property_file_full_path = ConcatPathComponent<PathChar>(property_folder_path, property_file_path);
-    LoadTensorProtoFromFile(property_file_full_path, property_protos, "[custom_property]");
+  InlinedHashMap<std::string, ONNX_NAMESPACE::TensorProto> param_tensor_protos;
 
-    for (auto& property_proto : property_protos) {
-      property_bag.AddProperty(property_proto);
+  const auto flatbuffer_tensors_to_tensor_protos = [&param_tensor_protos](const auto& flatbuffer_tensors) {
+    for (auto* fbs_tensor : flatbuffer_tensors) {
+      ORT_RETURN_IF_NOT(fbs_tensor, "Checkpoint is invalid. Expected: Valid flatbuffer tensor. Actual: nullptr.");
+      ONNX_NAMESPACE::TensorProto tensor_proto;
+      OrtFormatLoadOptions load_options{false, false};
+      ORT_RETURN_IF_ERROR(fbs::utils::LoadInitializerOrtFormat(*fbs_tensor, tensor_proto, load_options));
+      param_tensor_protos.insert({fbs_tensor->name()->str(), tensor_proto});
     }
-  }
 
-  return Status::OK();
-}
+    return Status::OK();
+  };
 
-Status OrtLoadInternal(const PathString& checkpoint_path,
-                       ONNX_NAMESPACE::ModelProto& model_proto) {
-  // Find tensor proto files.
-  InlinedHashMap<std::string, ONNX_NAMESPACE::TensorProto> param_tensor_protos;
-  InlinedVector<PathString> tensor_proto_filenames;
-
-  FilterFilesFromDirectory(
-      checkpoint_path,
-      [&tensor_proto_filenames](const PathChar* filename) -> bool {
-        PathString filename_str = filename;
-        if (StringEndsWith(filename_str, k_tensor_proto_file_name)) {
-          tensor_proto_filenames.push_back(filename_str);
-        }
-        return true;
-      });
-
-  // Load tensor protos to the tensorProto Vector
-  for (const auto& tensor_file_path : tensor_proto_filenames) {
-    std::vector<ONNX_NAMESPACE::TensorProto> tensor_protos{};
-    const auto tensor_file_full_path = ConcatPathComponent<PathChar>(checkpoint_path, tensor_file_path);
-    LoadTensorProtoFromFile(tensor_file_full_path, tensor_protos, "[params]");
+  const auto* requires_grad = module_state->requires_grad();
```

Comment:
```suggestion
  const auto* requires_grad = module_state->requires_grad();
```
The name looks not self-explained, is it possible we call it requires_grad(_params to match frozen_params ?